components/onboarding_tasks/onboarding_tasks_manager.tsx

```diff
@@ -39,65 +39,146 @@ import {
+import Clipboard from 'components/common/svg_images_components/clipboard_svg';
+import Gears from 'components/common/svg_images_components/gears_svg';
+import Handshake from 'components/common/svg_images_components/handshake_svg';
+import Phone from 'components/common/svg_images_components/phone_svg';
+import Security from 'components/common/svg_images_components/security_svg';
+import Sunglasses from 'components/common/svg_images_components/sunglasses_svg';
+import Wrench from 'components/common/svg_images_components/wrench_svg';
+
 import {OnboardingTaskCategory, OnboardingTaskList, OnboardingTasksName, TaskNameMapToSteps} from './constants';
+import {generateTelemetryTag} from './utils';
 
 const getCategory = makeGetCategory();
 
-const taskLabels = {
-    [OnboardingTasksName.CHANNELS_TOUR]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_learn_more_about_messaging'
-            defaultMessage='🔀 Learn about messaging'
-        />
-    ),
-    [OnboardingTasksName.BOARDS_TOUR]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.plan_sprint_with_kanban_style_boards'
-            defaultMessage='🎯 Plan a sprint with Kanban-style boards'
-        />),
-    [OnboardingTasksName.PLAYBOOKS_TOUR]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_resolve_incidents_faster_with_playbooks'
-            defaultMessage='🐛 Resolve incidents faster with playbooks'
-        />
-    ),
-    [OnboardingTasksName.INVITE_PEOPLE]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_invite_team_members'
-            defaultMessage='👋 Invite team members to the workspace'
-        />
-    ),
-    [OnboardingTasksName.COMPLETE_YOUR_PROFILE]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_complete_your_profile'
-            defaultMessage='📷 Complete your profile'
-        />
-    ),
-    [OnboardingTasksName.EXPLORE_OTHER_TOOLS]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.explore_other_tools_in_platform'
-            defaultMessage='⛰️ Explore other tools in the platform'
-        />
-    ),
-    [OnboardingTasksName.DOWNLOAD_APP]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_download_mm_apps'
-            defaultMessage='📱 Download the Desktop and Mobile Apps'
-        />
-    ),
-    [OnboardingTasksName.VISIT_SYSTEM_CONSOLE]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_visit_system_console'
-            defaultMessage='🥰️ Visit the System Console to configure your workspace'
-        />
-    ),
-    [OnboardingTasksName.START_TRIAL]: (
-        <FormattedMessage
-            id='onboardingTask.checklist.task_start_enterprise_trial'
-            defaultMessage='🏢 Learn more about Enterprise-level high-security features'
-        />
-    ),
+const useGetTaskDetails = () => {
```

Comment:
I think there was another comment regarding the `use...` in this function. It can cause confusion with hooks, maybe just `getTaskDetails`.

Reply:
I am using `useIntl` hook inside of this function. This function either needs to be a react component or a custom react hook hence the 'use' prefix.

Reply:
`TaskDetails` ?

Reply:
Then when I call it the lint complains that the name starting with uppercase letter should only be used as a constructor. 😄 
I think its Ok to be a hook as it calls a hook and calls it unconditionally.